src/python/python-proto/python_proto/pipeline.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import annotations
+
+import json
+import uuid
+
+from typing import cast
+
+from graplinc.grapl.pipeline.v1beta1.types_pb2 import Envelope as _Envelope
+from graplinc.grapl.pipeline.v1beta1.types_pb2 import Metadata as _Metadata
+from graplinc.grapl.pipeline.v1beta1.types_pb2 import Uuid as _Uuid
+
+from python_proto import SerDe
+
+class Uuid(SerDe):
+    lsb: int
+    msb: int
+
+    def __init__(self, lsb: int, msb: int) -> None:
+        self.lsb = lsb
+        self.msb = msb
+
+    @staticmethod
+    def deserialize(bytes_: bytes) -> Uuid:
+        uuid_ = _Uuid()
+        uuid_.parseFromString(bytes_)
+        return Uuid._from_proto(proto_uuid=uuid_)
+
+    @staticmethod
+    def from_uuid(uuid_: uuid.UUID) -> Uuid:
+        lsb_bytes = uuid_.bytes[:8]
+        msb_bytes = uuid_.bytes[8:]
+        lsb = int.from_bytes(lsb_bytes, byteorder="little", signed=False)
+        msb = int.from_bytes(msb_bytes, byteorder="little", signed=False)
+        return Uuid(lsb=lsb, msb=msb)
+
+    def into_uuid(self) -> uuid.UUID:
+        lsb_bytes = int.to_bytes(self.lsb, byteorder="little", length=8, signed=False)
+        msb_bytes = int.to_bytes(self.msb, byteorder="little", length=8, signed=False)
+        return uuid.UUID(bytes=lsb_bytes + msb_bytes)
+
+    def serialize(self) -> bytes:
+        return cast(bytes, self._into_proto().SerializeToString())
+
+    @staticmethod
+    def _from_proto(proto_uuid: _Uuid) -> Uuid:
+        return Uuid(lsb=proto_uuid.lsb, msb=proto_uuid.msb)
+
+    def _into_proto(self) -> _Uuid:
+        proto_uuid = _Uuid()
+        proto_uuid.lsb = self.lsb
+        proto_uuid.msb = self.msb
+        return proto_uuid
+
+    def __str__(self) -> str:
+        return str(self.into_uuid())
+
+    def __repr__(self) -> str:
+        return "Uuid " + json.dumps({"lsb": self.lsb, "msb": self.msb})
+
+    def __hash__(self) -> int:
+        return hash(self.into_uuid())
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Uuid):
+            return NotImplemented
+        return self.into_uuid() == other.into_uuid()
```

Comment:
food for thought:
if we @dataclass this, then I think we'd get all of this + init for free. Same with other SerDes

Reply:
I like this idea, will investigate

Reply:
done